File: app/src/main/java/br/com/pedrosilva/tecnonutri/presentation/ui/listeners/SingleOnClickListener.java
package br.com.pedrosilva.tecnonutri.presentation.ui.listeners;

import android.os.SystemClock;
import android.view.View;

/**
 * Created by psilva on 3/20/17.
 */
public abstract class SingleOnClickListener implements View.OnClickListener {
    private static long lastClickTime = 0;

    @Override
    public void onClick(View view) {
        if (SystemClock.elapsedRealtime() - lastClickTime < 500) {
            return;
        }
        lastClickTime = SystemClock.elapsedRealtime();
        onItemClick(view);
    }

    public abstract void onItemClick(View view);
}

File: app/src/main/kotlin/br/com/pedrosilva/tecnonutri/presentation/ui/listeners/SingleOnClickListener.kt
package br.com.pedrosilva.tecnonutri.presentation.ui.listeners

import android.os.SystemClock
import android.view.View

abstract class SingleOnClickListener : View.OnClickListener {

    companion object {
        private const val CLICK_INTERVAL_TIME = 500L
        private var lastClickTime = 0L
    }

    override fun onClick(view: View) {
        if (SystemClock.elapsedRealtime() - lastClickTime < CLICK_INTERVAL_TIME) {
            return
        }
        lastClickTime = SystemClock.elapsedRealtime()
        onItemClick(view)
    }

    abstract fun onItemClick(view: View)
}

fun View.setSingleOnClickListener(listener: (View) -> Unit) {
    setOnClickListener(object : SingleOnClickListener() {
        override fun onItemClick(view: View) {
            listener(view)
        }
    })
}